firestore/app/src/main/java/com/google/example/firestore/DocSnippets.java

```diff
@@ -1335,4 +1335,92 @@ public void onComplete(@NonNull Task<AggregateQuerySnapshot> task) {
         });
         // [END count_aggregate_query]
     }
+
+    public void orQuery() {
+        CollectionReference collection = db.collection("cities");
+        // [START or_queries]
+        Query query = collection.where(Filter.and(
+            Filter.greaterThan("name", "L"),
+            Filter.or(
+                Filter.equalTo("capital", true),
+                Filter.greaterThanOrEqualTo("population", 1000000)
+            )
+        ));
```

Comment:
This is an invalid query because it contains inequalities on multiple fields ("name" and "population").

Reply:
How's this one? I'll update it for all platforms.

```java
Query query = collection.where(Filter.and(
    Filter.equalTo("state", "CA"),
    Filter.or(
        Filter.equalTo("capital", true),
        Filter.greaterThanOrEqualTo("population", 1000000)
    )
));
```

Reply:
I thought about adding one snippet with just the `OR` part and then another one showing an `OR` with an `AND`.  Should we just show this one with `OR` and `AND`?